src/NServiceBus.Extensions.Hosting/IEndpointConfigurationBuilder.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace NServiceBus.Extensions.Hosting
+{
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Provides the ability to influence the endpoint configuration at the time when dependency injection is available.
+    /// </summary>
+    public interface IEndpointConfigurationBuilder
+    {
+        /// <summary>
+        ///
+        /// </summary>
+        /// <returns></returns>
+        Task<EndpointConfiguration> Build();
```

Comment:
Technically, we could also use the more modern ValueTask here but I wasn't sure how many people are familiar with this already and since this is not a hot path I didn't bother